core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/SyncMongoClientFactory.java

```diff
@@ -30,10 +30,13 @@ public static MongoClient getClient(
     if (dataDir == null) {
       throw new IllegalArgumentException("StitchAppClient not configured with a data directory");
     }
+
+    final String userId = appInfo.getAuthMonitor().isLoggedIn()
+        ? appInfo.getAuthMonitor().getActiveUserId() : "unbound";
     final String instanceKey = String.format(
```

Comment:
[q] do we even need the instance key at all anymore? now that we are storing data for different users in different data directories, I think this is unnecessary. getting rid of it will let users have longer db and coll names as well

Reply:
This instance key is actually only used in caching the clients in hashmap.

Reply:
ah okay, cool